### PR TITLE
Makefile: Suppress 'which' error messages

### PR DESCRIPTION
**Pull request by jwendell**

### CODE REVIEW

**Comment by mattdelco** [comment]
Out of curiousity, in what os/environment/config is there an error message?  Neither my Debian Linux nor Mac systems show any errors when a matching file can't be located.

**Comment by jwendell** [comment]
Fedora:
```sh
ABC=$(which non-existing-file || echo "DEF")
/usr/bin/which: no non-existing-file in (/usr/lib64/qt-3.3/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/home/jwendell/.local/bin:/home/jwendell/bin:/home/jwendell/go/src/github.com/openshift/origin/_output/local/bin/linux/amd64:/home/jwendell/go/bin:/home/jwendell/.local/bin:/home/jwendell/bin:/home/jwendell/go/src/github.com/openshift/origin/_output/local/bin/linux/amd64:/home/jwendell/go/bin)
[jwendell@jw-laptop ~]$ echo $ABC
DEF
```

File: Makefile [hunk 1]
```diff
@@ -122,8 +122,8 @@ ifeq ($(TAG),)
 endif
 
 # Discover if user has dep installed -- prefer that
-DEP    := $(shell which dep    || echo "${ISTIO_BIN}/dep" )
-GOLINT := $(shell which golint || echo "${ISTIO_BIN}/golint" )
+DEP    := $(shell which dep    2>/dev/null || echo "${ISTIO_BIN}/dep" )
+GOLINT := $(shell which golint 2>/dev/null || echo "${ISTIO_BIN}/golint" )
```
Comment (ldemailly) [inline Makefile]: in theory we should use command I think instead of which (but maybe not in a makefile)

**Comment by ldemailly** [comment]
unrelated circleci failures so I manually merged

**Pull request merged**